src/go/cmd/token-vendor/app/tokenvendor.go

```diff
@@ -95,33 +95,42 @@ func (tv *TokenVendor) GetOAuth2Token(ctx context.Context, jwtk string) (*tokens
 	return r, err
 }
 
-func (tv *TokenVendor) getOAuth2Token(ctx context.Context, jwtk string) (*tokensource.TokenResponse, error) {
+func (tv *TokenVendor) ValidateJWT(ctx context.Context, jwtk string) (string, error) {
 	p, err := jwt.PayloadUnsafe(jwtk)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to extract JWT payload")
+		return "", errors.Wrap(err, "failed to extract JWT payload")
 	}
 	exp := time.Unix(p.Exp, 0)
 	if exp.Before(time.Now()) {
-		return nil, fmt.Errorf("JWT has expired %v, %v ago (iss: %q)",
+		return "", fmt.Errorf("JWT has expired %v, %v ago (iss: %q)",
 			exp, time.Since(exp), p.Iss)
 	}
 	if err := acceptedAudience(p.Aud, tv.accAud); err != nil {
-		return nil, errors.Wrapf(err, "validation of JWT audience failed (iss: %q)", p.Iss)
+		return "", errors.Wrapf(err, "validation of JWT audience failed (iss: %q)", p.Iss)
 	}
 	if !IsValidDeviceID(p.Iss) {
-		return nil, fmt.Errorf("missing or invalid device identifier (`iss`: %q)", p.Iss)
+		return "", fmt.Errorf("missing or invalid device identifier (`iss`: %q)", p.Iss)
 	}
 	deviceID := p.Iss
 	pubKey, err := tv.repo.LookupKey(ctx, deviceID)
```

Comment:
Not blocking this CL, but this is another scaling concern: This lookup isn't cached so every websocket connection will eat into our GKE QPS.